backend/lab-service/src/main/java/org/mltooling/lab/services/AbstractServiceManager.java

```diff
@@ -323,14 +328,24 @@ public boolean deleteWorkspace(String user) throws Exception {
     return deleteService(workspaceService.getDockerName(), true, null);
   }
 
-  public LabService resetWorkspace(String user) throws Exception {
+  public LabService resetWorkspace(String user, String imageName) throws Exception {
     // remove workspace without volumes
-    shutdownWorkspace(user);
+    String previousUsedImage = shutdownWorkspace(user);
+
+    if (previousUsedImage == null){
+      previousUsedImage = CoreService.WORKSPACE.getImage();
+      log.info("Unable to retrieve previous used image");
+    }
+
+    if (imageName == null){
+      imageName = previousUsedImage;
+      log.info("Not image given, used previous used image: " + imageName);
```

Comment:
Do we need this log? If so I would make it a debug log and maybe reformulate to:
"No image provided, using previously used image."

Reply:
Changed in [781b1d0](https://github.com/SAP/machine-learning-lab/pull/26/commits/781b1d017d0f46a8bc3a18289b453e10309ebaa2)